examples/dmrg/2ddmrg.jl

```diff
@@ -10,13 +10,13 @@ let
 
   lattice = squareLattice(Nx,Ny,yperiodic=false)
 
-  ampo = AutoMPO(sites)
+  ampo = AutoMPO()
   for b in lattice
     add!(ampo,0.5,"S+",b.s1,"S-",b.s2)
     add!(ampo,0.5,"S-",b.s1,"S+",b.s2)
     add!(ampo,    "Sz",b.s1,"Sz",b.s2)
   end
-  H = toMPO(ampo)
+  H = toMPO(ampo,sites)
```

Comment:
I definitely prefer having the sites passed to the MPO constructor instead of to the AutoMPO! 

Should we consider to calling the constructor just `MPO` instead of `toMPO`? I think we only used `toMPO` in the C++ version because we wanted to pass keyword arguments to the constructor but couldn't because of C++.

Reply:
Yes, the only downside is giving a way to check if the AutoMPO input is out of bounds or not. But this does seem like the right move. 

I’m good will calling it MPO again yes! Let’s keep toMPO also to assist people already used to C++.

Reply:
I guess bounds checking can still be done when creating the MPO from the AutoMPO and the sites, right?

Reply:
Yes - ok good point! It doesn't hurt to delay it a little if it gets done eventually.

Also, in a way it gives more flexibility to AutoMPO. Like one could use sites past N for making infinite MPO's, then it's up to the conversion function to figure out whether it wants to allow that or not.